src/commonMain/kotlin/com/configcat/model/SettingValue.kt

```diff
@@ -0,0 +1,70 @@
+package com.configcat.model
+
+import com.configcat.Client.SettingTypeHelper.toSettingTypeOrNull
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Setting Value contains the proper value based on type.
+ */
+@Serializable
+public data class SettingValue(
+    @SerialName("b")
+    var booleanValue: Boolean? = null,
+
+    @SerialName("s")
+    var stringValue: String? = null,
+
+    @SerialName("i")
+    var integerValue: Int? = null,
+
+    @SerialName("d")
+    var doubleValue: Double? = null
+) {
+    public fun equalsBasedOnSettingType(other: Any?, settingType: Int): Boolean {
```

Comment:
This method serves a special internal purpose, I'd rather not expose it to the end users.

```suggestion
    internal fun equalsBasedOnSettingType(other: Any?, settingType: Int): Boolean {
```